runtime/compiler/control/JITaaSCompilationThread.hpp

```diff
@@ -65,6 +66,10 @@ class ClientSessionData
       uintptrj_t totalInstanceSize;
       PersistentUnorderedMap<int32_t, TR_OpaqueClassBlock *> *_classOfStaticCache;
       PersistentUnorderedMap<int32_t, TR_OpaqueClassBlock *> *_constantClassPoolCache;
+      PersistentUnorderedMap<int32_t, TR_J9MethodFieldAttributes> *_fieldAttributesCache;
+      PersistentUnorderedMap<int32_t, TR_J9MethodFieldAttributes> *_staticAttributesCache;
```

Comment:
I left that union in, because then we can use  `getCachedFieldAttributes` for both AOT and non-AOT. Otherwise, we'd need to have an if statement every time cache is accessed.

Reply:
But I do think that using unions for polymorphism is weird.
I think it's worth thinking more about it and implementing it in a different PR.